class.jetpack.php

```diff
@@ -587,6 +587,37 @@ private function __construct() {
 			add_action( 'wp_print_styles', array( $this, 'implode_frontend_css' ), -1 ); // Run first
 			add_action( 'wp_print_footer_scripts', array( $this, 'implode_frontend_css' ), -1 ); // Run first to trigger before `print_late_styles`
 		}
+
+		// Check if site icon is available in core, and if so convert Jetpack's to use it.
+		add_action( 'admin_init', array( 'Jetpack', 'jetpack_site_icon_available_in_core' ) );
```

Comment:
Is there a better hook than `admin_init`? Or a better place to call it, like in `Jetpack_Site_icon` itself?
Currently it would add that callback for every admin page load for(almost)ever.

Same goes for the following hooks.